setuptools/config/setupcfg.py

```diff
@@ -582,11 +582,11 @@ def _parse_version(self, value):
             # accidentally include newlines and other unintended content
             try:
                 Version(version)
-            except InvalidVersion:
+            except InvalidVersion as e:
                 raise OptionError(
                     f'Version loaded from {value} does not '
                     f'comply with PEP 440: {version}'
-                )
+                ) from e
```

Comment:
Do we actually want to have the previous exception also shown?

Would the increased traceback bring more information for the user to debug, or is the current error message enough?

The problem with `from None` is that it also reduces the stack trace isn't it? I think that probably the way it is written right now gets the sweet spot: enough stack trace for debugging, but no extra unnecessary verboseness. Maybe it is just bugbear being overly pedantic...

Reply:
For future reference: see resolution in https://github.com/pypa/setuptools/pull/4168#discussion_r1440649502.